sklearn/feature_selection/_multivariate_selection.py

```diff
@@ -0,0 +1,214 @@
+    operation occurs in which a multivariate independence test 
+    is performed for each data matrix with the selected best 
+    features and an additional feature not yet selected. The 
+    additional feature associated with the highest multivariate
+    independence test statistic is then chosen as the next best 
+    feature. 
+    
+    Read more in the :ref:`User Guide <multivariate_feature_selection>`.
+    
+    Parameters
+    ----------
+    k: int, default=10
+        amount of features to select. 
+        
+    Attributes
+    ----------
+    features_ : array, shape (n_features,)
+        indices of all features in X
+    
+    best_features_ : array, shape (k,)
+         indices of selected k best features of features_
+         
+    Examples
+    --------
+    >>> from sklearn.datasets import load_digits
+    >>> from sklearn.feature_selection import MultivariateFeatureSelector
+    >>> X, y = load_digits(return_X_y=True)
+    >>> X.shape
+    (1797, 64)
+    >>> X_new = MultivariateFeatureSelector(k = 7).fit_transform(X, y)
+    >>> X_new.shape
+    (1797, 7)
+    """
+    
+    def __init__(self, k=10):
+        self.k = k
+        
+    class _Parallel:
+        #helper class for calculating, in parallel, 
+        #test statistic associated with
+        #selected best features and an additional feature 
+        def __init__(self, X_new, y,best_features):
+            self.X_new = X_new
+            self.y = y
+            self.best_feat = best_features
+
+        def __call__(self, index):
+            if np.var(self.X_new[:,index]) == 0:
+                stat = -1000.0
+            else:   
+                if len(self.best_feat)==0:
+                    X_j =  self.X_new[:,index] 
+                    stat= k_sample_test(X_j,self.y)
+                else:
+                    columns = self.best_feat.copy() 
+                    columns.append(index)
+                    X_j = self.X_new[:,columns]
+                    stat= k_sample_test(X_j,self.y)
+            return stat
+        
+        
+    def fit(self, X, y,workers = -1):
+        """Learn the features to select from X.
+        
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training vectors, where `n_samples` is the number of samples and
+            `n_features` is the number of predictors.
+        y : array-like of shape (n_samples,), default=None
+            Target values. 
+        
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        if isspmatrix(X) == True:
+            X = X.toarray()
+        
+        features = np.arange(X.shape[1])
+        
+        if np.isnan(X).any() == True:
+            raise ValueError("existing non-parametric multivariate independence tests in scipy do not allow nan")
+        if type(self.k) != int:
+            raise ValueError("k features to select must be integer")
```

Comment:
This is a `TypeError`

Reply:
Rewrite as `k is type {}, must be int` where `{}` is `type(k)`

Reply:
what exactly do you want in these brackets, because I have currently copied the brackets directly, but then as the ValueError occurs, brackets show but it seems like there should be user input inside, like it says float if its a float

Reply:
So the string should be something like `"{}".format(x)` where the value of `x` will show up inside the bracket

Reply:
https://github.com/srecupe2/scikit-learn/pull/4/commits/184e5432f44d112873543cc2c7e0ae22e2ebb3f7